sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -264,12 +281,12 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         df,
         sink,
         outputMode,
-        useTempCheckpointLocation = true,
+        useTempCheckpointLocation = isTempCheckpointLocationAvailable,
         trigger = trigger)
     } else {
       val (useTempCheckpointLocation, recoverFromCheckpointLocation) =
         if (source == "console") {
-          (true, false)
+          (isTempCheckpointLocationAvailable, false)
```

Comment:
Well, actually I don't think "java.io.tmpdir" will ever be on a filesystem different from the local one. But, the other PR forces the metadata to be written on the local filesystem, despite the default one is different (for instance it can be HDFS).
This means that in a distributed environment, which should be fault tolerant, with that patch  if a node fails we loose the metadata. Since one of the involved sink is the `foreach` one, which can be used to write the data somewhere (for example HBase or Kafka), I think that forcing the user to specify a `checkpointLocation` which is created on the `defaultFs` (in this case HDFS) would be a better option.